bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/discovery/VehicleDiscovery.java

```diff
@@ -12,158 +12,63 @@
-                                    + Constants.SEMICOLON);
-                        } else {
-                            remoteServicesDisabled
-                                    .append(RemoteServiceHandler.RemoteService.CLIMATE_NOW_START.getLabel()
-                                            + Constants.SEMICOLON);
-                        }
-                        properties.put("remoteServicesEnabled", remoteServicesEnabled.toString().trim());
-                        properties.put("remoteServicesDisabled", remoteServicesDisabled.toString().trim());
-
-                        // Update Properties for already created Things
-                        bridge.getThing().getThings().forEach(vehicleThing -> {
-                            Configuration c = vehicleThing.getConfiguration();
-                            if (c.containsKey(MyBMWConstants.VIN)) {
-                                String thingVIN = c.get(MyBMWConstants.VIN).toString();
-                                if (vehicle.vin.equals(thingVIN)) {
-                                    vehicleThing.setProperties(properties);
-                                }
-                            }
-                        });
+    private static final int DISCOVERY_TIMEOUT = 10;
 
-                        // Properties needed for functional Thing
-                        properties.put(MyBMWConstants.VIN, vehicle.vin);
-                        properties.put("vehicleBrand", vehicle.brand);
-                        properties.put("refreshInterval",
-                                Integer.toString(MyBMWConstants.DEFAULT_REFRESH_INTERVAL_MINUTES));
+    private Optional<MyBMWBridgeHandler> bridgeHandler = Optional.empty();
+    private Optional<MyBMWProxy> myBMWProxy = Optional.empty();
+    private Optional<ThingUID> bridgeUid = Optional.empty();
 
-                        String vehicleLabel = vehicle.brand + " " + vehicle.model;
-                        Map<String, Object> convertedProperties = new HashMap<String, Object>(properties);
-                        thingDiscovered(DiscoveryResultBuilder.create(uid).withBridge(bridgeUID)
-                                .withRepresentationProperty(MyBMWConstants.VIN).withLabel(vehicleLabel)
-                                .withProperties(convertedProperties).build());
-                    }
-                });
-            });
-        });
+    public VehicleDiscovery() {
+        super(MyBMWConstants.SUPPORTED_THING_SET, DISCOVERY_TIMEOUT, false);
     }
 
     @Override
     public void setThingHandler(ThingHandler handler) {
-        if (handler instanceof MyBMWBridgeHandler bmwBridgeHandler) {
-            bridgeHandler = Optional.of(bmwBridgeHandler);
-            bridgeHandler.get().setDiscoveryService(this);
+        if (handler instanceof MyBMWBridgeHandler) {
+            logger.trace("VehicleDiscovery.setThingHandler for MybmwBridge");
+            bridgeHandler = Optional.of((MyBMWBridgeHandler) handler);
```

Comment:
Why did you change this?

Reply:
I don't remember, had some issues. I reverted it.